src/generator_p.h

```diff
@@ -109,7 +109,7 @@ struct Generator::GeneratorImpl
 
     std::string generateMethodBodyCode(const std::string &methodBody) const;
 
-    std::string generateDoubleCode(const std::string &value) const;
+    static std::string generateDoubleCode(const std::string &value);
```

Comment:
As an alternative this method could be made a function and then the `Analyzer` class wouldn't need to be a friend of the `Generator class`.

Reply:
Agreed. See 35c8b0c7.